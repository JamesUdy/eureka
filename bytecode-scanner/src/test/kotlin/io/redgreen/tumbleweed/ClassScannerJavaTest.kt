package io.redgreen.tumbleweed

import io.redgreen.tumbleweed.samples.LambdaAccessingField
import io.redgreen.tumbleweed.samples.MethodReadingField
import io.redgreen.tumbleweed.samples.MethodWritingField
import org.approvaltests.Approvals
import org.junit.jupiter.api.Test

class ClassScannerJavaTest {
  private val defaultJavaClassLocation = ClassFileLocation(
    compiledClassesDirectory = "../bytecode-samples/build/classes/java/main",
    fqClassName = "replace-java-class-?"
  )

  @Test
  fun `it can scan a class with methods reading a field`() {
    // given
    val methodReadingField = defaultJavaClassLocation.copy(
      fqClassName = MethodReadingField::class.java.name,
    )

    // when
    val classStructure = ClassScanner.scan(methodReadingField.file)

    // then
    Approvals.verify(classStructure.printable)
  }

  @Test
  fun `it can scan a class with methods writing a field`() {
    // given
    val methodWritingField = defaultJavaClassLocation.copy(
      fqClassName = MethodWritingField::class.java.name,
    )

    // when
    val classStructure = ClassScanner.scan(methodWritingField.file)

    // then
    Approvals.verify(classStructure.printable)
  }

  @Test
  fun `it can scan a class with a lambda function accessing a field`() {
    // given
    val methodReadingAndWritingField = defaultJavaClassLocation.copy(
      fqClassName = LambdaAccessingField::class.java.name,
    )

    // when
    val classStructure = ClassScanner.scan(methodReadingAndWritingField.file)

    // then
    Approvals.verify(classStructure.printable)
  }
}
