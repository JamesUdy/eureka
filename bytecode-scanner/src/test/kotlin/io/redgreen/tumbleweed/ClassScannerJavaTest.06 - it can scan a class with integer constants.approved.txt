Package: io.redgreen.tumbleweed.samples
Class: IntegerConstants
Extends: java.lang.Object
Fields:
  - int MINUS_4096
  - int MINUS_ONE
  - int ZERO
  - int ONE
  - int TWO
  - int THREE
  - int FOUR
  - int FIVE
  - int SIX
Methods:
  - void <init>()
  - void main([]java.lang.String)
Reads:
  - void main([]java.lang.String) -> int MINUS_4096
  - void main([]java.lang.String) -> int MINUS_ONE
  - void main([]java.lang.String) -> int ZERO
  - void main([]java.lang.String) -> int ONE
  - void main([]java.lang.String) -> int TWO
  - void main([]java.lang.String) -> int THREE
  - void main([]java.lang.String) -> int FOUR
  - void main([]java.lang.String) -> int FIVE
  - void main([]java.lang.String) -> int SIX
